hugo/content/guides/multiple-languages.md

```diff
@@ -0,0 +1,454 @@
+
+```ts
+const fileSystemWatcher = vscode.workspace.createFileSystemWatcher('**/*.(me|who|hello)');
+context.subscriptions.push(fileSystemWatcher);
+
+// Options to control the language client
+const clientOptions: LanguageClientOptions = {
+    documentSelector: [
+        { scheme: 'file', language: 'multiple-languages-configuration' },
+        { scheme: 'file', language: 'multiple-languages-definition' },
+        { scheme: 'file', language: 'multiple-languages-implementation' }
+    ],
+    synchronize: {
+        // Notify the server about file changes to files contained in the workspace
+        fileEvents: fileSystemWatcher
+    }
+};
+```
+
+> Be aware of that the file extensions can change over time, depending on your needs. You should actually use the `LanguageMetaData` of each language service. I have chosen the given implementation because of simplicity. You can do it better!
+
+## Test the extension!
+
+Now everything should be executable. **Do not forget to build**!
+
+Let's run the extension and create some files in our workspace:
+
+### Definition `people.who`
+
+```
+person Markus
+person Michael
+person Frank
+```
+
+### Configuration `thats.me`
+
+```
+I am Markus.
+```
+
+### Implementation `greetings.hello`
+
+```
+Hello Markus!
+Hello Michael!
+```
+
+## Checklist
+
+You should be able now...:
+* to see proper syntax highlighting
+* to trigger auto completion for keywords
+* to jump to the definitin by Cmd/Ctrl-clicking on a person's name
+
+# Add a validator (task)
+
+As promised, let's add a simple validation rule, that you cannot greet yourself. Therefore we enter our name in the `thats.me` file like we did in the previous step.
+
+Try to include the follwing code to our validator. This is meant as task, try to find the missing pieces on your own :-).
+
+```ts
+checkNotGreetingYourself(greeting: Greeting, accept: ValidationAcceptor): void {
+    const document = getDocument(greeting);
+    const configFilePath = join(document.uri.fsPath, '..', 'thats.me');
+    const configDocument = this.documents.getOrCreateDocument(URI.file(configFilePath));
+    if (greeting.person.ref) {
+        if (configDocument && isConfigurationUnit(configDocument.parseResult.value)) {
+            if(configDocument.parseResult.value.who.ref === greeting.person.ref) {
+                accept('warning', 'You cannot greet yourself 🙄!', { node: greeting, property: 'person' });
+            }
+        }
+    }
+}
+```
+
+After doing so, your name should be have a warning, stating that you cannot greet yourself.
+
+# Troubleshooting
+
+In this section we will list common mistakes.
+
+* One prominent mistake is forgetting to build Langium and Typescript files, before running the extension.
+
+* Since we are basically just copy-pasting given configuration, be aware what you are pasting. Make sure that the code still makes sense after copying. Probably you forgot to adapt the pasted code.
```

Comment:
```suggestion
* Since we are basically just copy-pasting given configuration, be aware of what you are pasting. Make sure that the code still makes sense after copying. You probably forgot to adapt the pasted code.
```